### PR TITLE
 snowflake_java based apis

### PR DESCRIPTION
**Pull request by navinrathore**
Replaced with snowflake_java api. Cleaner code.  Now have vararg (Column ..) based functions

However, below are the few things which are need to be resolveed

a) filter() -> as arg two columns are compared by equalTo(Column) ..
    earlier used $eq$eq$eq,  but in java api no equal_to(Object) function
    .in(Object) is the nearest one but there is no not_in(Object) function
b) Row().getAs("ColumnName") fn is not there rather Row().get(Index) is present....Need to get convert from column index
c) DF.Util()    GraphFrame is spark based
d) persist(), rapartition(0 etc. are WIP for snowpark

### CODE REVIEW

File: core/src/main/java/zingg/Labeller.java [hunk 1]
```diff
@@ -95,11 +96,11 @@ public void processRecordsCli(DataFrame lines) throws ZinggClientException {
 			int totalPairs = clusterIDs.size();
 			
 			for (int index = 0; index < totalPairs; index++){	
-				DataFrame currentPair = lines.filter(lines.col(ColName.CLUSTER_COLUMN).in(
-						clusterIDs.get(index).getAs(ColName.CLUSTER_COLUMN))).cacheResult();
+				DataFrame currentPair = lines.filter(lines.col(ColName.CLUSTER_COLUMN).equal_to(
+					Functions.lit(clusterIDs.get(index).get(0)))).cacheResult();
```
Comment (sonalgoyal) [inline core/src/main/java/zingg/Labeller.java:100]: We should refer to column names only, let us not use index for retrieval as it is very brittle

File: core/src/main/java/zingg/Trainer.java [hunk 2]
```diff
@@ -35,26 +35,26 @@ public void execute() throws ZinggClientException {
...
+			DataFrame positives = null;
+			DataFrame negatives = null;
+			DataFrame tra = DSUtil.getTraining(snow, args);
 			tra = DSUtil.joinWithItself(tra, ColName.CLUSTER_COLUMN, true);
-			tra = tra.cache();
-			positives = tra.filter(tra.col(ColName.MATCH_FLAG_COL).equalTo(ColValues.MATCH_TYPE_MATCH));
-			negatives = tra.filter(tra.col(ColName.MATCH_FLAG_COL).equalTo(ColValues.MATCH_TYPE_NOT_A_MATCH));
+			tra = tra.cacheResult();
+			positives = tra.filter(tra.col(ColName.MATCH_FLAG_COL).in(ColValues.MATCH_TYPE_MATCH));
+			negatives = tra.filter(tra.col(ColName.MATCH_FLAG_COL).in(ColValues.MATCH_TYPE_NOT_A_MATCH));
```
Comment (sonalgoyal) [inline core/src/main/java/zingg/Trainer.java:44]: why not functions.lit ?

File: core/src/main/java/zingg/hash/First3CharsBox.java [hunk 3]
```diff
@@ -37,8 +37,8 @@ public Integer call(String field) {
 	 }
 	 
 	 @Override
-	 public Object apply(Row ds, String column) {
-		 return call((String) ds.getAs(column));
+	 public Object apply(Row ds, int column) {
+		 return call((String) ds.get(column));
```
Comment (sonalgoyal) [inline core/src/main/java/zingg/hash/First3CharsBox.java:41]: let us discuss this

File: core/src/main/java/zingg/hash/IsNullOrEmpty.java [hunk 4]
```diff
@@ -15,7 +15,7 @@ public Boolean call(String field) {
 		 return (field == null || ((String ) field).trim().length() == 0);
 	 }
 
-	public Object apply(Row ds, String column) {
-		 return call((String) ds.getAs(column));
+	public Object apply(Row ds, int column) {
+		 return call((String) ds.get(column));
```
Comment (sonalgoyal) [inline core/src/main/java/zingg/hash/IsNullOrEmpty.java:19]: to discuss

File: core/src/main/java/zingg/util/HashUtil.java [hunk 5]
```diff
@@ -37,7 +38,7 @@
 				});
 		for (HashFnFromConf scriptArg : scriptArgs) {
 			HashFunction fn = HashFunctionRegistry.getFunction(scriptArg.getName());
-			spark.udf().register(fn.getName(), (UDF1) fn, fn.getReturnType());
+			snow.udf().registerPermanent(fn.getName(), (JavaUDF1<Vector, Double>) fn, fn.getDataType(), fn.getReturnType(), "stageLocation");
```
Comment (sonalgoyal) [inline core/src/main/java/zingg/util/HashUtil.java:41]: parameter type of JavaUDF - why is it a vector?

File: core/src/main/java/zingg/util/Metric.java [hunk 6]
```diff
@@ -16,7 +16,7 @@
     public static final long timeout = 1200L;
     public static final double confidence = 0.95; // default value
 
-    public static double approxCount(Dataset<Row> data) {
-        return data.rdd().countApprox(timeout, confidence).initialValue().mean();
+    public static double approxCount(DataFrame data) {
+        return data.async().count().getResult((int)timeout/1000);
```
Comment (sonalgoyal) [inline core/src/main/java/zingg/util/Metric.java:20]: please comment this out

**Review by sonalgoyal — CHANGES_REQUESTED** [review]

File: core/src/main/java/zingg/util/Metric.java (hunk 6 above)
Comment (navinrathore) [reply core/src/main/java/zingg/util/Metric.java:20]: Excluded through pom.xml
Comment this function as well

File: core/src/main/java/zingg/util/HashUtil.java (hunk 5 above)
Comment (navinrathore) [reply core/src/main/java/zingg/util/HashUtil.java:41]: remove parameters

File: core/src/main/java/zingg/Trainer.java (hunk 2 above)
Comment (navinrathore) [reply core/src/main/java/zingg/Trainer.java:44]: done

File: core/src/main/java/zingg/Labeller.java (hunk 1 above)
Comment (navinrathore) [reply core/src/main/java/zingg/Labeller.java:100]: changed in the best manner. Will revisit if we have general solution of Row elements extraction.

**Pull request closed without merging**